remove "NPY_NO_DEPRECATED_API", "NPY_1_7_API_VERSION" from setup_package.py files. 
author: John Dey Fred Hutch Cancer Center Seattle WA USA
diff -ruN astropy-6.0.1.orig/astropy/convolution/setup_package.py astropy-6.0.1/astropy/convolution/setup_package.py
--- astropy-6.0.1.orig/astropy/convolution/setup_package.py	2024-03-26 11:10:49.000000000 -0700
+++ astropy-6.0.1/astropy/convolution/setup_package.py	2025-02-03 16:40:21.399979000 -0800
@@ -22,7 +22,7 @@
     ]
     _convolve_ext = Extension(
         name="astropy.convolution._convolve",
-        define_macros=[("NPY_NO_DEPRECATED_API", "NPY_1_7_API_VERSION")],
+        # define_macros=[("NPY_NO_DEPRECATED_API", "NPY_1_7_API_VERSION")],
         extra_compile_args=extra_compile_args,
         include_dirs=[numpy.get_include()],
         sources=sources,
diff -ruN astropy-6.0.1.orig/astropy/io/ascii/setup_package.py astropy-6.0.1/astropy/io/ascii/setup_package.py
--- astropy-6.0.1.orig/astropy/io/ascii/setup_package.py	2024-03-26 11:10:49.000000000 -0700
+++ astropy-6.0.1/astropy/io/ascii/setup_package.py	2025-02-03 16:42:51.840821000 -0800
@@ -15,7 +15,7 @@
     ]
     ascii_ext = Extension(
         name="astropy.io.ascii.cparser",
-        define_macros=[("NPY_NO_DEPRECATED_API", "NPY_1_7_API_VERSION")],
+        # define_macros=[("NPY_NO_DEPRECATED_API", "NPY_1_7_API_VERSION")],
         include_dirs=[numpy.get_include()],
         sources=sources,
     )
diff -ruN astropy-6.0.1.orig/astropy/stats/setup_package.py astropy-6.0.1/astropy/stats/setup_package.py
--- astropy-6.0.1.orig/astropy/stats/setup_package.py	2024-03-26 11:10:49.000000000 -0700
+++ astropy-6.0.1/astropy/stats/setup_package.py	2025-02-03 16:42:26.248842000 -0800
@@ -13,14 +13,14 @@
 def get_extensions():
     _sigma_clip_ext = Extension(
         name="astropy.stats._fast_sigma_clip",
-        define_macros=[("NPY_NO_DEPRECATED_API", "NPY_1_7_API_VERSION")],
+        # define_macros=[("NPY_NO_DEPRECATED_API", "NPY_1_7_API_VERSION")],
         sources=SRCFILES,
         include_dirs=[numpy.get_include()],
         language="c",
     )
     _stats_ext = Extension(
         name="astropy.stats._stats",
-        define_macros=[("NPY_NO_DEPRECATED_API", "NPY_1_7_API_VERSION")],
+        # define_macros=[("NPY_NO_DEPRECATED_API", "NPY_1_7_API_VERSION")],
         sources=[os.path.join(ROOT, "_stats.pyx")],
         include_dirs=[numpy.get_include()],
     )
diff -ruN astropy-6.0.1.orig/astropy/table/setup_package.py astropy-6.0.1/astropy/table/setup_package.py
--- astropy-6.0.1.orig/astropy/table/setup_package.py	2024-03-26 11:10:49.000000000 -0700
+++ astropy-6.0.1/astropy/table/setup_package.py	2025-02-03 16:32:48.226409000 -0800
@@ -15,7 +15,7 @@
     exts = [
         Extension(
             name="astropy.table." + os.path.splitext(source)[0],
-            define_macros=[("NPY_NO_DEPRECATED_API", "NPY_1_7_API_VERSION")],
+            # define_macros=[("NPY_NO_DEPRECATED_API", "NPY_1_7_API_VERSION")],
             sources=[os.path.join(ROOT, source)],
             include_dirs=include_dirs,
         )
diff -ruN astropy-6.0.1.orig/astropy/timeseries/periodograms/bls/setup_package.py astropy-6.0.1/astropy/timeseries/periodograms/bls/setup_package.py
--- astropy-6.0.1.orig/astropy/timeseries/periodograms/bls/setup_package.py	2024-03-26 11:10:49.000000000 -0700
+++ astropy-6.0.1/astropy/timeseries/periodograms/bls/setup_package.py	2025-02-03 16:41:55.178171000 -0800
@@ -11,7 +11,7 @@
 def get_extensions():
     ext = Extension(
         "astropy.timeseries.periodograms.bls._impl",
-        define_macros=[("NPY_NO_DEPRECATED_API", "NPY_1_7_API_VERSION")],
+        # define_macros=[("NPY_NO_DEPRECATED_API", "NPY_1_7_API_VERSION")],
         sources=[
             os.path.join(BLS_ROOT, "bls.c"),
             os.path.join(BLS_ROOT, "_impl.pyx"),
diff -ruN astropy-6.0.1.orig/astropy/timeseries/periodograms/lombscargle/setup_package.py astropy-6.0.1/astropy/timeseries/periodograms/lombscargle/setup_package.py
--- astropy-6.0.1.orig/astropy/timeseries/periodograms/lombscargle/setup_package.py	2024-03-26 11:10:49.000000000 -0700
+++ astropy-6.0.1/astropy/timeseries/periodograms/lombscargle/setup_package.py	2025-02-03 16:41:36.953665000 -0800
@@ -11,7 +11,7 @@
 def get_extensions():
     ext = Extension(
         "astropy.timeseries.periodograms.lombscargle.implementations.cython_impl",
-        define_macros=[("NPY_NO_DEPRECATED_API", "NPY_1_7_API_VERSION")],
+        # define_macros=[("NPY_NO_DEPRECATED_API", "NPY_1_7_API_VERSION")],
         sources=[os.path.join(ROOT, "implementations", "cython_impl.pyx")],
         include_dirs=[numpy.get_include()],
     )
diff -ruN astropy-6.0.1.orig/setup.py astropy-6.0.1/setup.py
--- astropy-6.0.1.orig/setup.py	2024-03-26 11:10:49.000000000 -0700
+++ astropy-6.0.1/setup.py	2025-02-03 16:19:10.190859000 -0800
@@ -71,6 +71,6 @@
 # Specify the minimum version for the Numpy C-API
 for ext in ext_modules:
     if ext.include_dirs and "numpy" in ext.include_dirs[0]:
-        ext.define_macros.append(("NPY_TARGET_VERSION", "NPY_1_21_API_VERSION"))
+        ext.define_macros.append(("NPY_TARGET_VERSION", "NPY_1_25_API_VERSION"))
 
 setup(ext_modules=ext_modules)
